src/DynamoCoreWpf/Views/CodeBlocks/CodeBlockEditor.xaml.cs

```diff
@@ -383,10 +384,15 @@ private void ShowInsightWindow(IEnumerable<CodeBlockInsightItem> items)
         /// <param name="e"></param>
         void TextArea_LostFocus(object sender, RoutedEventArgs e)
         {
-            this.InnerTextEditor.TextArea.ClearSelection();
-            this.nodeViewModel.DynamoViewModel.ExecuteCommand(
-                   new DynCmd.UpdateModelValueCommand(
-                       this.nodeViewModel.NodeModel.GUID, "Code", this.InnerTextEditor.Text));           
+            InnerTextEditor.TextArea.ClearSelection();
+            var recorder = nodeViewModel.WorkspaceViewModel.Model.UndoRecorder;
+
+            if (string.IsNullOrEmpty(InnerTextEditor.Text))
+                DiscardChangesAndOptionallyRemoveNode(recorder);
+            else
+                CommitChanges(recorder);
```

Comment:
Thanks @pboyer, I have heard you said it before. It was my personal preference to leave them out for single-line statement, but it's about time I took this feedback seriously. Will start doing that from next time on.